homeassistant/components/recollect_waste/sensor.py

```diff
@@ -84,12 +86,19 @@ def icon(self):
     def update(self):
         """Update device state."""
         try:
-            pickup_event = self.client.get_next_pickup()
+            pickup_event_array = self.client.get_pickup_events(
+                date.today(), date.today() + timedelta(weeks=4)
```

Comment:
Because the original function, get_next_pickup, does exactly that under the covers in the python module:
https://github.com/stealthhacker/python-recollect-waste/blob/master/recollect_waste/__init__.py